gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -447,6 +450,72 @@ gst_tensor_filter_init (GstTensorFilter * self)
   gst_tensors_config_init (&self->out_config);
 }
 
+/**
+ * @brief Clear and reset data.
+ */
+static void
+gst_tensor_filter_reset (GstTensorFilter * self)
+{
+  GstTensorFilterProperties *prop;
+  guint i;
+
+  prop = &self->prop;
+
+  gst_tensor_filter_close_fw (self);
+
+  if (prop->fwname) {
+    g_free ((char *) prop->fwname);
```

Comment:
This comment might be trivial: 
1. the parameter of g_free can be nullable thus you do not need to be check whether the argument is null or not.
2. In order to free const pointer, casting it to take off its const attribute is general to me. However, if you enable -Wcast-qual, the compiler said that "error: cast discards ‘const’ qualifier from pointer target type".  If you want to avoid this issue, simply use the following macro: 
```suggestion
   #define g_free_const(x) g_free((void*)(long)(x))
    g_free_const (prop->fwname);
```

Reply:
Thanks for the advice about the inline function. it's applied :)